File: main/electronics-shop/electronics-shop.cs
using System;

internal static class Solution {
    private static int[] GetRecord()
            => Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
    
    private static void Main() // naive brute-force algorithm (n * m <= 10**6)
    {
        var s = GetRecord()[0]; // don't need n, m
        var kbds = GetRecord();
        var usbs = GetRecord();
        
        var best = -1;
        
        foreach (var kbd in kbds) {
            foreach (var usb in usbs) {
                var cur = kbd + usb;
                if (best < cur && cur <= s) best = cur;
            }
        }
        
        Console.WriteLine(best);
    }
}

File: main/electronics-shop/electronics-shop.pl
#!/usr/bin/env perl

use strict;
use warnings;

sub read_record {
    return split / /ms, <>;
}

my ($cap) = read_record; # need s as $cap, but don't need n, m
my @kbds = read_record;
my @usbs = read_record;

my $best = -1;

for my $kbd (@kbds) {
    for my $usb (@usbs) {
        my $cur = $kbd + $usb;
        if ($best < $cur && $cur <= $cap) { $best = $cur }
    }
}

print $best, "\n";
